src/hobbits-plugins/importerexporters/UsbDevice/usbdevice.h

```diff
@@ -0,0 +1,70 @@
+#pragma once
+
+#include "importexportinterface.h"
+#include "parameterdelegate.h"
+#include <libusb-1.0/libusb.h>
+
+class UsbDevice : public QObject, ImporterExporterInterface
+{
+    Q_OBJECT
+    Q_PLUGIN_METADATA(IID "hobbits.ImporterExporterInterface.UsbDevice")
+    Q_INTERFACES(ImporterExporterInterface)
+
+public:
+    UsbDevice();
+
+    ImporterExporterInterface* createDefaultImporterExporter() override;
+
+    QString name() override;
+    QString description() override;
+    QStringList tags() override;
+
+    bool canExport() override;
+    bool canImport() override;
+
+    virtual QSharedPointer<ParameterDelegate> importParameterDelegate() override;
+    virtual QSharedPointer<ParameterDelegate> exportParameterDelegate() override;
+
+    QSharedPointer<ImportResult> importBits(const Parameters &parameters,
+                                            QSharedPointer<PluginActionProgress> progress) override;
+
+
+    QSharedPointer<ExportResult> exportBits(QSharedPointer<const BitContainer> container,
+                                            const Parameters &parameters,
+                                            QSharedPointer<PluginActionProgress> progress) override;
+
+   struct usbParams setupLibusb(usbParams params);
```

Comment:
instead of passing in and returning the structure, you can use pass-by-reference to avoid the copying: https://en.cppreference.com/w/cpp/language/reference